enverus_direct_access/__init__.py

```diff
@@ -523,3 +442,154 @@ def query(self, dataset, **options):
 
             for record in records:
                 yield record
+
+            if self.links is None:
+                break
+
+
+class DirectAccessV2(BaseAPI):
+    """Client for Enverus' Developer API Version 2"""
+
+    def __init__(
+            self,
+            client_id,
+            client_secret,
+            api_key,
+            retries=5,
+            backoff_factor=1,
+            links=None,
+            access_token=None,
+            **kwargs
+    ):
+        """
+        Enverus' Developer API Version 2 client
+
+        API documentation and credentials can be found at: https://app.enverus.com/direct/#/api/explorer/v2/gettingStarted
+
+        :param client_id: client id credential.
+        :type client_id: str
+        :param client_secret: client secret credential.
+        :type client_secret: str
+        :param api_key: api key credential.
+        :type api_key: str
+        :param retries: the number of attempts when retrying failed requests with status codes of 500, 502, 503 or 504
+        :type retries: int
+        :param backoff_factor: the factor to use when exponentially backing off prior to retrying a failed request
+        :type backoff_factor: int
+        :param links: a dictionary of prev and next links as provided by the python-requests Session.
+        See https://requests.readthedocs.io/en/master/user/advanced/#link-headers
+        :type dict
+        :param access_token: an optional, pregenerated access token. If provided, the class instance will not
+        automatically try to request a new access token.
+        :type: access_token: str
+        :param kwargs:
+        """
+        super(DirectAccessV2, self).__init__(api_key, retries, backoff_factor, **kwargs)
+        self.client_id = client_id
+        self.client_secret = client_secret
+        self.links = links
+        self.access_token = access_token
+        self.url = self.url + "/v2/direct-access"
+        self.session.headers.update({"X-API-KEY": self.api_key})
+        self.session.hooks["response"].append(self._check_response)
+
+        if self.access_token:
+            self.session.headers["Authorization"] = "bearer {}".format(
+                self.access_token
+            )
+        else:
+            self.access_token = self.get_access_token()["access_token"]
+
+    def get_access_token(self):
+        """
+        Get an access token from /tokens endpoint. Automatically sets the Authorization header on the class instance's
+        session. Raises DAAuthException on error
+
+        :return: token response as dict
+        """
+        url = self.url + "/tokens"
```

Comment:
Do not use "+" for string concatenation, use os.path.join for url parts concatenation